components/forms/Form/Form.tsx

```diff
@@ -45,6 +49,39 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues> & DynamicForm
     }
   }, [formStatusError, errorList, lastSubmitCount, canFocusOnError]);
 
+  useEffect(() => {
+    // calculate initial delay for submit timer
+    const secondsBaseDelay = 2;
+    const secondsPerFormElement = 2;
+    const numberOfRequiredElements = React.Children.toArray(children).filter((child) => {
+      try {
```

Comment:
I would not know (right now) how to simplify this code but I have this feeling that a cleaner solution could exist 😕 
Maybe someone else in the team will propose something 👀